libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.cpp

```diff
@@ -0,0 +1,659 @@
+#include <AP_Logger/AP_Logger.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_HAL/utility/sparse-endian.h>
+#include <AP_Common/Bitmask.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+
+extern const AP_HAL::HAL &hal;
+
+// constructor
+AP_ExternalAHRS_InertialLabs::AP_ExternalAHRS_InertialLabs(AP_ExternalAHRS *_frontend,
+                                                           AP_ExternalAHRS::state_t &_state) :
+    AP_ExternalAHRS_backend(_frontend, _state)
+{
+    auto &sm = AP::serialmanager();
+    uart = sm.find_serial(AP_SerialManager::SerialProtocol_AHRS, 0);
+    if (!uart) {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "InertialLabs ExternalAHRS no UART");
+        return;
+    }
+    baudrate = sm.find_baudrate(AP_SerialManager::SerialProtocol_AHRS, 0);
+    port_num = sm.find_portnum(AP_SerialManager::SerialProtocol_AHRS, 0);
+
+    // don't offer IMU by default, at 200Hz it is too slow for many aircraft
+    set_default_sensors(uint16_t(AP_ExternalAHRS::AvailableSensor::GPS) |
+                        uint16_t(AP_ExternalAHRS::AvailableSensor::BARO) |
+                        uint16_t(AP_ExternalAHRS::AvailableSensor::COMPASS));
+    
+    if (!hal.scheduler->thread_create(FUNCTOR_BIND_MEMBER(&AP_ExternalAHRS_InertialLabs::update_thread, void), "ILabs", 2048, AP_HAL::Scheduler::PRIORITY_SPI, 0)) {
+        AP_HAL::panic("InertialLabs Failed to start ExternalAHRS update thread");
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "InertialLabs ExternalAHRS initialised");
+}
+
+/*
+  re-sync buffer on parse failure
+ */
+void AP_ExternalAHRS_InertialLabs::re_sync(void)
+{
+    if (buffer_ofs > 3) {
+        /*
+          look for the 2 byte header and try to sync to that
+         */
+        const uint16_t header = 0x55AA;
+        const uint8_t *p = (const uint8_t *)memmem(&buffer[1], buffer_ofs-3, &header, sizeof(header));
+        if (p != nullptr) {
+            const uint16_t n = p - &buffer[0];
+            memmove(&buffer[0], p, buffer_ofs - n);
+            buffer_ofs -= n;
+        } else {
+            buffer_ofs = 0;
+        }
+    } else {
+        buffer_ofs = 0;
+    }
+}
+
+// macro for checking we don't run past end of message buffer
+#define CHECK_SIZE(d) need_re_sync = (message_ofs + (msg_len=sizeof(d)) > buffer_end); if (need_re_sync) break
+
+// lookup a message in the msg_types bitmask to see if we received it in this packet
+#define GOT_MSG(mtype) msg_types.get(unsigned(MessageType::mtype))
+
+/*
+  check header is valid
+ */
+bool AP_ExternalAHRS_InertialLabs::check_header(const ILabsHeader *h) const
+{
+    return h->magic == 0x55AA &&
+        h->msg_type == 1 &&
+        h->msg_id == 0x95 &&
+        h->msg_len <= sizeof(buffer)-2;
+}
+
+/*
+  check the UART for more data
+  returns true if we have consumed potentially valid bytes
+ */
+bool AP_ExternalAHRS_InertialLabs::check_uart()
+{
+    WITH_SEMAPHORE(state.sem);
+
+    if (!setup_complete) {
+        return false;
+    }
```

Comment:
In the constructor, if the uart failed, you get a log message of `InertialLabs ExternalAHRS no UART`, but it proceeds. 
`uart` would be nullptr. 

Thus, I think should add a uart nulltpr check here to prevent null pointer dereference.

Reply:
setup_complete can never go true unless we have the uart, as it only gets set if the thread starts. I'm pretty sure this is fine